### PR TITLE
optimized automatic positioning of form check in issue detail 

### PR DESCRIPTION
**Pull request by laojun**
## What type of PR is this?

- [x] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## What this PR does / why we need it:
1：optimized automatic positioning of form check in issue detail 
2：fix ref in fixed select

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [x] Yes(screenshot is required)
- [] No

## Special notes for your reviewer:
Kapture 2021-05-13 at 11 20 44

### CODE REVIEW

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx [hunk 1]
```diff
@@ -389,29 +417,29 @@ const IssueMetaFields = ({ labels, isEditMode, isBacklog, editAuth, issueType, f
...
-    ...insertWhen(expandCustomFields && issueType === ISSUE_TYPE.TASK, [
+    ...insertWhen(issueType === ISSUE_TYPE.TASK, [
       {
         className: 'mb20 full-width',
         name: 'taskType',
         label: i18n.t('task type'),
         type: 'select',
         showRequiredMark: true,
-        itemProps: { options: taskTypeOptions, allowClear: false },
+        itemProps: { options: taskTypeOptions, allowClear: false, hidden: !expandCustomFields },
```
Comment (daskyrk) [inline shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:427]: Pass `className` prop is better.

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx [hunk 2]
```diff
@@ -128,6 +128,29 @@ const IssueMetaFields = ({ labels, isEditMode, isBacklog, editAuth, issueType, f
     ));
   }, [taskTypeList]);
 
+  React.useEffect(()=>{
+    if(ref && !ref.current){
+      const customFieldKeys = map(customFieldDetail?.property || [], 'propertyName').concat(['taskType', 'bugStage', 'owner']);
```
Comment (McDaddy) [inline shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:133]: no need `|| []`,    `map(undefined, 'xxx')` output is `[]`

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx [hunk 3]
```diff
@@ -128,6 +128,29 @@ const IssueMetaFields = ({ labels, isEditMode, isBacklog, editAuth, issueType, f
...
+      ref.current = {
+        onFocus: (fKey: string) => {
+          const curRef = ref.current?.refMap[fKey];
+          if(fKey === 'issueManHour.elapsedTime'){
+            // 点击打开时间跟踪
```
Comment (McDaddy) [inline shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:138]: if it's new written code, better comment by english

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx [hunk 4]
```diff
@@ -128,6 +128,29 @@ const IssueMetaFields = ({ labels, isEditMode, isBacklog, editAuth, issueType, f
...
+            curRef.click();
+          } else if(customFieldKeys.includes(fKey)){
+            //需要打开自定义
+            setExpandCustomFields(true)
+            setTimeout(()=>{
+              curRef?.focus && curRef.focus();
```
Comment (McDaddy) [inline shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:144]: `curRef?.focus()` directly

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx [hunk 5]
```diff
@@ -589,6 +625,7 @@ export const EditIssueDrawer = (props: IProps) => {
   const { addIssueRelation } = issueStore.effects;
   const { updateCustomFieldDetail } = issueStore.reducers;
   const { id: orgID } = orgStore.useStore(s => s.currentOrg);
+  const metaFieldsRef = React.useRef(null);
```
Comment (McDaddy) [inline shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:628]: define generic type for this ref?

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx (hunk 2 above)
Comment (laojun) [reply shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:133]: to prevention ```customFieldDetail.property = null```

File: shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx (hunk 4 above)
Comment (laojun) [reply shell/app/modules/project/common/components/issue/edit-issue-drawer.tsx:144]: could get a error when curRef is exist and focus does't exist;

**Review by daskyrk — APPROVED** [review]

**Pull request merged**